src/Components/Masonry.tsx

```diff
@@ -0,0 +1,19 @@
+import { Box } from "@artsy/palette"
+import styled from "styled-components"
+import { GridColumnGapProps, gridColumnGap, style } from "styled-system"
+
+const columnCount = style({
+  prop: "columnCount",
+  cssProperty: "columnCount",
+})
+
+export const Masonry = styled(Box)<
+  { columnCount: number[] | number } & GridColumnGapProps
+>`
+  ${columnCount};
+  ${gridColumnGap};
+
+  > * {
+    break-inside: avoid;
+  }
```

Comment:
So I stole this from your ConsignFAQ — I didn't replace it though because of this break-inside addition. Let me know if you want me to use this there as well.

Reply:
Yeah! i wanna keep exploring this for artworks, see if there's something we can hammer in.

Would you mind adding a storybooks entry for `Masonry`?

Reply:
![](http://static.damonzucconi.com/_capture/HbBJKE1CGYdM.png)

Made the sort obvious — was thinking if we really wanted to maintain the left-to-right sorting we could partition the children into columned arrays then re-interleave the array. Probably overkill but not the worst idea. 🤷

Reply:
Yeah, we actually _need_ to do that. If you scroll fast down the artwork grid, or click an artwork further down the grid while in the experiment a/b test group you ocan see that our current grid is vertically oriented -- it leads to a flash of the lazy loader, even though those items are within view, because the indexes aren't correctly left to right, and `preload` exists only for the first ten. Its a mess.

Reply:
![flash](https://user-images.githubusercontent.com/236943/81990750-a02bb800-95f4-11ea-961e-8a3597e4faa9.gif)

Reply:
(cc @zephraph - as you have the most context around this)